File: resize_icons.js
const fs = require('fs');
const path = require('path');

// Simple Node.js script to copy logo as placeholder for icons
// In a real environment, you'd use sharp or canvas for resizing

console.log('🎨 Creating PWA icons from logo.png...');

try {
  // Check if logo exists
  const logoPath = 'public/icons/logo.png.png';
  if (!fs.existsSync(logoPath)) {
    console.error('❌ logo.png.png not found');
    process.exit(1);
  }

  console.log('✅ Found logo.png.png');
  
  // Create icon files (in real scenario, these would be resized)
  const iconSizes = [
    { name: 'icon-192x192.png', width: 192, height: 192 },
    { name: 'icon-512x512.png', width: 512, height: 512 },
    { name: 'apple-touch-icon.png', width: 180, height: 180 }
  ];

  iconSizes.forEach(icon => {
    // For now, copy the logo as placeholder
    // In production, replace this with actual resized images
    fs.copyFileSync(logoPath, `public/icons/${icon.name}`);
    console.log(`✅ Created ${icon.name} (${icon.width}x${icon.height})`);
  });

  console.log('\n📱 PWA icons ready!');
  console.log('⚠️  Note: These are copies of your logo. Resize them to the specified dimensions for optimal display.');
  
} catch (error) {
  console.error('❌ Error:', error.message);
}

File: resize_icons.py
#!/usr/bin/env python3
"""
Resize logo.png to PWA icon sizes using Pillow
"""

from PIL import Image
import os

def resize_logo():
    try:
        # Open the original logo
        logo_path = "public/icons/logo.png.png"
        if not os.path.exists(logo_path):
            print(f"Error: {logo_path} not found")
            return
        
        with Image.open(logo_path) as img:
            print(f"Original logo size: {img.size}")
            
            # Create icons directory if it doesn't exist
            os.makedirs("public/icons", exist_ok=True)
            
            # Resize to 192x192
            img_192 = img.resize((192, 192), Image.Resampling.LANCZOS)
            img_192.save("public/icons/icon-192x192.png", "PNG", optimize=True)
            print("✅ Created icon-192x192.png (192x192)")
            
            # Resize to 512x512
            img_512 = img.resize((512, 512), Image.Resampling.LANCZOS)
            img_512.save("public/icons/icon-512x512.png", "PNG", optimize=True)
            print("✅ Created icon-512x512.png (512x512)")
            
            # Resize to 180x180 (Apple touch icon)
            img_180 = img.resize((180, 180), Image.Resampling.LANCZOS)
            img_180.save("public/icons/apple-touch-icon.png", "PNG", optimize=True)
            print("✅ Created apple-touch-icon.png (180x180)")
            
            print("\n🎨 All PWA icons created successfully!")
            print("📱 Ready for mobile installation!")
            
    except Exception as e:
        print(f"❌ Error: {e}")
        print("Make sure Pillow is installed: pip install Pillow")

if __name__ == "__main__":
    resize_logo()
